Source/svg/SVGParser.swift

```diff
@@ -254,15 +255,43 @@ open class SVGParser {
             }
         }
     }
-
+    
     fileprivate func parseElement(_ node: XMLIndexer, groupStyle: [String: String] = [:]) -> Node? {
         guard let element = node.element else { return .none }
-
-        let styleAttributes = getStyleAttributes(groupStyle, element: element)
-        if styleAttributes["display"] == "none" {
+        
+        let nodeStyle = getStyleAttributes(groupStyle, element: element)
+        if nodeStyle["display"] == "none" {
             return .none
         }
-
+        
+        guard let parsedNode = parseElementInternal(node, groupStyle: nodeStyle) else { return .none }
+        
+        if let filterString = element.allAttributes["filter"]?.text ?? nodeStyle["filter"], let shadowId = parseIdFromUrl(filterString), let shadow = defShadows[shadowId] {
+            if shadow.offset == nil {
+                parsedNode.effect = shadow.input
+                return parsedNode
+            }
+            
+            guard let offset = shadow.offset else { return .none }
+            let shadowNode = parseElementInternal(node, groupStyle: nodeStyle)! // TODO copy
```

Comment:
Does this TODO require some more work?

Reply:
A lot. Have to implement copy for all the fields of all the of all the node types.